src/tools/api.js

```diff
@@ -47,9 +47,18 @@ async function callForSearch({ query, client }) {
         }],
     });
 
+    const { results } = response[0].data.Typeahead_autocomplete;
+
+    let locationId;
+    for (let i = 0; i < results.length; i++) {
+        if (!locationId && results[i].locationId) {
+            locationId = results[i].locationId;
+        }
+    }
```

Comment:
Cleaner alternative that ends quickly

```
const locationId = results.find((result) => !!result.locationId)
```

Reply:
@metalwarrior665
Honestly, I think that my implementation is more readable and understandable)
And iterable array is tiny, so anyway it will be quick.

But I can change it to your variant. Shall I?
And thanks for tip)

Reply:
I'm not here to force my way, I should be able to persuade you if needed :)

1. I think `for (let i = 0` should only be used if you need to use the index variable. Otherwise, we should use `for (const ... of ...)` because it is simpler. You don't need to think if the below code will use the `i` and the access `result` is simpler than `results[i]`. Do you agree?
Now we have 
```javascript
let locationId;
    for (const result of results.length) {
        if (!locationId && result.locationId) {
            locationId = result.locationId;
        }
    }
```
2. Here you need to think about the extra `if (!locationId` part that is only there because we don't finish early. So easier would be
```javascript
let locationId;
    for (const result of results.length) {
        if ( result.locationId) {
            locationId = result.locationId;
            break;
        }
    }
```
Here you can clearly see that we end up whenever we found the first result.

3. And since JS gives you some declarative utilities for these common use-cases, we can use them. And that is what  `array.find` is exactly for. There you just declare that you want to find the first thing that matches and you even get rid of the `let`

Still not convinced?